File: src/dataset/javascript/binary-search_exercism-javascript-ThomasZumsteg/claude/ClaudeSonnet4_binary-search_v3.js
function BinarySearch(items) {
  this.items = items;
  this._isSorted = null;
}

BinarySearch.prototype = {
  get array() {
    if (this._isSorted !== null) {
      return this._isSorted ? this.items : undefined;
    }
    
    const len = this.items.length;
    for (let i = 1; i < len; i++) {
      if (this.items[i] < this.items[i - 1]) {
        this._isSorted = false;
        return undefined;
      }
    }
    
    this._isSorted = true;
    return this.items;
  },

  indexOf: function(item) {
    if (!this.array) return -1;
    
    let min = 0;
    let max = this.items.length - 1;
    
    while (min <= max) {
      const mid = (min + max) >>> 1;
      const midVal = this.items[mid];
      
      if (midVal < item) {
        min = mid + 1;
      } else if (midVal > item) {
        max = mid - 1;
      } else {
        return mid;
      }
    }
    
    return -1;
  }
};

export default BinarySearch;

File: src/dataset/c/binary_search_Exercism_vlzware/claude/ClaudeSonnet4_binary_search_v3.c
#include "binary_search.h"
#include <stddef.h>

int* binary_search(const int x, const int arr[], const int len)
{
    if (arr == NULL || len <= 0)
        return NULL;

    int down = 0;
    int up = len - 1;
    
    while (down <= up) {
        int mid = down + ((up - down) >> 1);
        int mid_val = arr[mid];
        
        if (x == mid_val)
            return (int *) &arr[mid];
        
        if (x > mid_val)
            down = mid + 1;
        else
            up = mid - 1;
    }

    return NULL;
}